internal/controlplane/translator/gateway.go

```diff
@@ -1 +1,91 @@
 package translator
+
+import (
+	"fmt"
+
+	v1 "github.com/api7/api7-ingress-controller/api/dashboard/v1"
+	"github.com/api7/api7-ingress-controller/pkg/id"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	gatewayv1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+func (t *Translator) TranslateGateway(tctx *TranslateContext, obj *gatewayv1.Gateway) (*TranslateResult, error) {
+	result := &TranslateResult{}
+	for _, listener := range obj.Spec.Listeners {
+		tctx.GatewayTLSConfig = append(tctx.GatewayTLSConfig, *listener.TLS)
+		ssl, err := t.translateSecret(tctx, listener, obj.Name, obj.Namespace)
+		if err != nil {
+			return nil, fmt.Errorf("failed to translate secret: %w", err)
+		}
+		result.SSL = append(result.SSL, ssl)
+	}
+	return result, nil
+}
+
+func (t *Translator) translateSecret(tctx *TranslateContext, listener gatewayv1.Listener, name, ns string) (*v1.Ssl, error) {
+	if tctx.Secrets == nil {
+		return nil, nil
+	}
+	sslObj := &v1.Ssl{}
+	sslObj.ID = id.GenID(fmt.Sprintf("%s_%s", ns, name))
+	if listener.Hostname != nil && *listener.Hostname != "" {
+		sslObj.Snis = []string{string(*listener.Hostname)}
+	}
+	secret := tctx.Secrets[types.NamespacedName{Namespace: ns, Name: name}]
+	cert, key, err := extractKeyPair(secret, true)
+	if err != nil {
+		return nil, err
+	}
+	sslObj.Cert = string(cert)
+	sslObj.Key = string(key)
+	sslObj.Labels = map[string]string{
```

Comment:
https://github.com/api7/api7-ingress-controller/blob/release-v2-dev/internal/controlplane/label/label.go#L10

It can be used to identify which gateway manages this SSL and can also be used for matching and searching.

Reply:
fixed